core/primitives/src/epoch_manager.rs

```diff
@@ -49,117 +40,977 @@ pub struct EpochConfig {
+                validator_to_index,
+                block_producers_settlement,
+                chunk_producers_settlement,
+                hidden_validators_settlement,
+                stake_change,
+                validator_reward,
+                validator_kickout,
+                fishermen_to_index,
+                minted_amount,
+                seat_price,
+                protocol_version,
+            })
+        }
+
+        #[inline]
+        pub fn epoch_height_mut(&mut self) -> &mut EpochHeight {
+            match self {
+                Self::V1(v1) => &mut v1.epoch_height,
+                Self::V2(v2) => &mut v2.epoch_height,
+            }
+        }
+
+        #[inline]
+        pub fn epoch_height(&self) -> EpochHeight {
+            match self {
+                Self::V1(v1) => v1.epoch_height,
+                Self::V2(v2) => v2.epoch_height,
+            }
+        }
+
+        #[inline]
+        pub fn seat_price(&self) -> Balance {
+            match self {
+                Self::V1(v1) => v1.seat_price,
+                Self::V2(v2) => v2.seat_price,
+            }
+        }
+
+        #[inline]
+        pub fn minted_amount(&self) -> Balance {
+            match self {
+                Self::V1(v1) => v1.minted_amount,
+                Self::V2(v2) => v2.minted_amount,
+            }
+        }
+
+        #[inline]
+        pub fn block_producers_settlement(&self) -> &[ValidatorId] {
+            match self {
+                Self::V1(v1) => &v1.block_producers_settlement,
+                Self::V2(v2) => &v2.block_producers_settlement,
+            }
+        }
+
+        #[inline]
+        pub fn chunk_producers_settlement(&self) -> &[Vec<ValidatorId>] {
+            match self {
+                Self::V1(v1) => &v1.chunk_producers_settlement,
+                Self::V2(v2) => &v2.chunk_producers_settlement,
+            }
+        }
+
+        #[inline]
+        pub fn validator_kickout(&self) -> &HashMap<AccountId, ValidatorKickoutReason> {
+            match self {
+                Self::V1(v1) => &v1.validator_kickout,
+                Self::V2(v2) => &v2.validator_kickout,
+            }
+        }
+
+        #[inline]
+        pub fn protocol_version(&self) -> ProtocolVersion {
+            match self {
+                Self::V1(v1) => v1.protocol_version,
+                Self::V2(v2) => v2.protocol_version,
+            }
+        }
+
+        #[inline]
+        pub fn stake_change(&self) -> &BTreeMap<AccountId, Balance> {
+            match self {
+                Self::V1(v1) => &v1.stake_change,
+                Self::V2(v2) => &v2.stake_change,
+            }
+        }
+
+        #[inline]
+        pub fn validator_reward(&self) -> &HashMap<AccountId, Balance> {
+            match self {
+                Self::V1(v1) => &v1.validator_reward,
+                Self::V2(v2) => &v2.validator_reward,
+            }
+        }
+
+        pub fn validators_iter(&self) -> ValidatorStakeIter {
```

Comment:
should these and following match v1/v2 methods be inline too?

Reply:
Yes, good catch. Added missing inlines in https://github.com/near/nearcore/pull/4115/commits/0af94dcc7eafef6304c2612f0aea35e2f29bb3b0